shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java

```diff
@@ -351,17 +331,13 @@ public AcquisitionAttributes<Long> getDicomAttributesForAcquisition(DatasetAcqui
 		LOG.debug("get DICOM attributes for acquisition " + acquisition.getId() + " : " + (new Date().getTime() - ts) + " ms");
 		return dAcquisitionAttributes;
 	}
-	
-	static String extractInstanceUID(String url) {
-		boolean condition1 = url != null && url.contains("objectUID=");
-		boolean condition2 = url != null && url.contains("instances/");
-		if (condition1) {
-			String[] split = StringUtils.splitByWholeSeparator(url, "objectUID=", 2);
-			return StringUtils.split(split[1], "&", 1)[0];
-		} else if (condition2) {
-			String[] split = StringUtils.splitByWholeSeparator(url, "instances/", 2);
-			return StringUtils.split(split[1], "/", 1)[0];
-		} else return null;
+
+	public WADOURLHandler getWadoURLHandler() {
```

Comment:
Do you really need this if it is autowired ?

Reply:
Unfortunately, yes! as the ExaminationAttributes class of Julien is a none-spring component and it requires the extract code as well, I found it more elegant to init the wado-url-handler by Spring into downloader and get it from there instead passing it through the methods.

Reply:
Could it be a static class at this moment ?